runtime/server/x86/CMakeLists.txt

```diff
@@ -150,6 +151,26 @@ add_library(decoder STATIC
 )
 target_link_libraries(decoder PUBLIC ${TORCH_LIBRARIES} kaldi-decoder utils post_processor)
 
+if(JEMALLOC)
+  set(jemalloc_SOURCE_DIR ${fc_base}/jemalloc-src CACHE PATH "jemalloc source directory")
+  set(jemalloc_BINARY_DIR ${fc_base}/jemalloc-build CACHE PATH "jemalloc build directory")
+  set(jemalloc_SUBBUILD_DIR ${fc_base}/jemalloc-subbuild CACHE PATH "jemalloc subbuild directory")
+  set(jemalloc_PREFIX_DIR ${jemalloc_SUBBUILD_DIR}/jemalloc-populate-prefix CACHE PATH "jemalloc prefix directory")
+  ExternalProject_Add(jemalloc
+    URL               https://github.com/jemalloc/jemalloc/releases/download/5.2.1/jemalloc-5.2.1.tar.bz2
+    URL_HASH          SHA256=34330e5ce276099e2e8950d9335db5a875689a4c6a56751ef3b1d8c537f887f6
+    TMP_DIR           ${jemalloc_SUBBUILD_DIR}/tmp
+    STAMP_DIR         ${jemalloc_SUBBUILD_DIR}/src/jemalloc-stamp
+    DOWNLOAD_DIR      ${jemalloc_SUBBUILD_DIR}/src
+    SOURCE_DIR        ${jemalloc_SOURCE_DIR}
+    BINARY_DIR        ${jemalloc_BINARY_DIR}
+    INSTALL_DIR       ${jemalloc_PREFIX_DIR}
+    CONFIGURE_COMMAND ${jemalloc_SOURCE_DIR}/configure --prefix=${jemalloc_PREFIX_DIR}
+    BUILD_COMMAND     make -j 4
+  )
+  target_link_libraries(decoder PUBLIC ${jemalloc_PREFIX_DIR}/lib/libjemalloc.a)
```

Comment:
Hi, there may be something wrong here. An error occer while I am building the project using command `mkdir build && cd build && cmake -DJEMALLOC=on .. && cmake --build .` in directory of `/home/wenet/runtime/server/x86`


The Error like this:
`/usr/bin/make -f CMakeFiles/label_checker_main.dir/build.make CMakeFiles/label_checker_main.dir/depend
make[2]: Entering directory '/home/wenet/runtime/server/x86/build'
cd /home/wenet/runtime/server/x86/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/wenet/runtime/server/x86 /home/wenet/runtime/server/x86 /home/wenet/runtime/server/x86/build /home/wenet/runtime/server/x86/build /home/wenet/runtime/server/x86/build/CMakeFiles/label_checker_main.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/wenet/runtime/server/x86/build'
/usr/bin/make -f CMakeFiles/label_checker_main.dir/build.make CMakeFiles/label_checker_main.dir/build
make[2]: Entering directory '/home/wenet/runtime/server/x86/build'
make[2]: *** No rule to make target '../fc_base/jemalloc-subbuild/jemalloc-populate-prefix/lib/libjemalloc.a', needed by 'label_checker_main'.  Stop.
make[2]: Leaving directory '/home/wenet/runtime/server/x86/build'
make[1]: *** [CMakeFiles/Makefile2:331: CMakeFiles/label_checker_main.dir/all] Error 2
make[1]: Leaving directory '/home/wenet/runtime/server/x86/build'
make: *** [Makefile:155: all] Error 2`


My environment is a docker container coping from Image "wenetorg/wenet-mini:latest".


I found the directory "fc_base/jemalloc-build" and "fc_base/jemalloc-subbuild" are all empty except "." and "..", so I guess the libary "jemalloc" is definitely not compiled.
Then, I try to replace `target_link_libraries(decoder PUBLIC ${jemalloc_PREFIX_DIR}/lib/libjemalloc.a)` with `add_dependencies(decoder jemalloc)`, it's working.

The reason may be "the dependency is not set up correctly".

In addition, It may be a problem worth considering that using library like "xxxx.a", because "xxxx.a" is not windows compatible.